docs/usage.rst

```diff
@@ -378,6 +378,30 @@ input Spec.
    Spec types, you should wrap the mapping Specs with ``s.merge`` and pass that to
    ``s.all``.
 
+.. _key_value_specs:
+
+Key/Value Specs
+^^^^^^^^^^^^^^^
+
+Mapping Specs are useful for heterogeneous associative data structures for which the
+keys are known *a priori*. However, you may often wish to validate a homogeneous
+mapping with unknown keys. For such cases, you can turn to
+:py:meth:`dataspec.SpecAPI.kv`.
+
+.. code-block:: python
+
+   spec = s.kv(s.str(regex=r"[A-Z]{2}"), s.str(regex=r"[A-Z][\w ]+"))
+   spec.is_valid({"GA": "Georgia", "NM": "New Mexico"})  # True
+   spec.is_valid({"ga": "Georgia", "NM": "New Mexico"})  # False
+   spec.is_valid({"ga": "Georgia", "NM": "new mexico"})  # False
+
+.. note::
+
+   By default :py:meth:`s.kv <dataspec.SpecAPI.kv>` will not conform keys on input
+   values, to avoid potential creating potentially duplicate keys from the key
+   conformer. You can override this behavior with the ``conform_keys`` keyword
+   argument.
```

Comment:
I don't quite understand this. Are you worried that conformation could potentially cause duplicate keys? How likely is this to happen?

Reply:
Hmm... I'm not very concerned about this. I think it should be an option, but I went back and forth whether or not to make it `False` by default. I landed on the more conservative option. I could be convinced otherwise though.

Reply:
especially now after looking at your rename spec I like the behavior there where you can specify if you want it to error out when your overwrite an existing key. If that was a default here and by default it used a conformer like all other specs then that would be nice :) but I'm not too worried either. just nice to be consistent.

Reply:
The more I think about it, the more I think this is probably not the direction _this_ particular Spec should go. `s.rename` (#68 for those following along at home) is a peculiar case (and one I've fretted about nearly constantly since I put it up) since it is primarily a conformer, not a validator. I added the validator for that as a kind of thought experiment and I'm still unsure how I feel about it.

In this case, I think it makes sense to just err on the side of caution by not conforming keys and let the user override the default if they think it's safe to do so. We're still in a very early version, so if we find this is not the desired behavior in most cases we can change make a breaking change. We could also think about adding that overwritten key validator later if it ends up being useful and desirable.

My guess though is that this will primarily end up being used as a kind of collection validator where the collection is key/value pairs and the primary validation for the keys will probably be something like "is this a string?" or "is this a string in a certain format?".